src/main/java/org/fungover/thunder/Client.java

```diff
@@ -0,0 +1,60 @@
+package org.fungover.thunder;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class Client {
+    private final String clientId;
+    private final TopicHandler topicHandler;
+    private boolean connected;
+    private Set<String> subscribedTopics;
+
+    public Client(String clientId, TopicHandler topicHandler) {
+        this.clientId = clientId;
+        this.topicHandler = topicHandler;
+        this.connected = false;
+        this.subscribedTopics = new HashSet<>();
+    }
+
+    public String getClientId() {
+        return clientId;
+    }
+
+    public boolean isConnected() {
+        return connected;
+    }
+
+    public void connect() {
+        if (!connected) {
+            // Establish a connection to the MQTT broker
+            // Perform the MQTT connection handshake
+            System.out.println("Client " + clientId + " connected to broker.");
+            connected = true;
+        } else {
+            System.out.println("Client " + clientId + " is already connected.");
+        }
+    }
```

Comment:
I do not really understand this. If a client is not connected, doesn't removeDisconnectedClients() in ClientHandler remove them from the list? Which would make this unnecessary?

Reply:
This is for checking whethere the client is connected or not. Just because it isn't connected, it doesn't disappear -- but it does get removed from the list of connected clients in the ClientHandler class. At least I thougt it works like this, but my networking skills aren't exactly great, I am kind of struggeling :-)

Reply:
Makes sense, this one helps to know when the client is ready to receive data(that is connected or not) and the other one is to keep an accurate list of active clients, right? Do we have any code yet that monitor the clients status? Thinking of reconnection, I have a vague memory of Martin mentioning something about if a user is out in the woods and looses connection to the broker, that the system can try to reconnect automatically. 🤔

Reply:
> This is for checking whethere the client is connected or not. Just because it isn't connected, it doesn't disappear -- but it does get removed from the list of connected clients in the ClientHandler class. At least I thougt it works like this, but my networking skills aren't exactly great, I am kind of struggeling :-)

I wasnt thinking that we could save disconnected clients so I misunderstood the usage. Well done emmelie.